src/libs/actions/Link.ts

```diff
@@ -64,6 +66,41 @@ function openOldDotLink(url: string) {
     );
 }
 
+function buildTravelDotURL(spotnanaToken?: string, postLoginPath?: string): Promise<string> {
+    const authCode = spotnanaToken ? `authCode=${spotnanaToken}` : '';
+    const tmcID = `tmcId=${CONST.SPOTNANA_TMC_ID}`;
+    const redirectURL = postLoginPath ? `redirectUrl=${postLoginPath}` : '';
+
+    const paramsArray = [authCode, tmcID, redirectURL];
+    const params = paramsArray.filter(Boolean).join('&');
+
+    return Environment.getTravelDotEnvironmentURL().then((environmentURL) => {
+        const travelDotDomain = Url.addTrailingForwardSlash(environmentURL);
+        return `${travelDotDomain}auth/code?${params}`;
+    });
+}
+
+/**
+ * @param postLoginPath When provided, we will redirect the user to this path post login on travelDot. eg: 'trips/:tripID'
+ */
+function openTravelDotLink(policyID: OnyxEntry<string>, postLoginPath?: string) {
+    if (policyID === null) {
+        return;
+    }
+
+    const parameters: GenerateSpotnanaTokenParams = {
+        policyID,
+    };
+
+    asyncOpenURL(
+        // eslint-disable-next-line rulesdir/no-api-side-effects-method
+        API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.GENERATE_SPOTNANA_TOKEN, parameters, {})
```

Comment:
> For commands where the network response must be accessed directly or when there is functionality that can only
> * happen once the request is finished (eg. calling third-party services like Onfido and Plaid, redirecting a user
> * depending on the response data, etc.).


you can see the docs. this is it's intended use

Reply:
So in this case, we are actually not calling a third party, this is a token generated by our server. That being said we do need to wait for the response before we can take any actions in the app, so I believe that the usage is still correct (since we are redirecting the user)